### PR TITLE
Fix all no else warnings

### PR DESCRIPTION
**Pull request by jenningsje**

### CODE REVIEW

**Comment by BoPeng** [comment]
Please complete #1468 before working on this because this PR includes content from #1468. If you work in parallel, fork this branch from master, not the branch for #1468.

File: src/sos/workflow_executor.py [hunk 1]
```diff
@@ -1662,7 +1658,8 @@ def run_as_master(self, targets=None, mode=None) -> Dict[str, Any]:
         )
 
         if exec_error.errors:
-            pending_steps = dag.pending()[1]
+            pending_steps = dag.pending()
```
Comment (BoPeng) [inline src/sos/workflow_executor.py:1661]: This is another failed merge.

File: src/sos/workflow_executor.py [hunk 2]
```diff
@@ -1945,7 +1939,7 @@ def run_as_nested(self,
                 exec_error.append(
                     self.workflow.name,
                     RuntimeError(
-                        f'{len(sections)} pending step{"s" if len(sections) > 1 else ""}: {", ".join(sections)}'
+                        f'{len(sections)} failed step{"s" if len(sections) > 1 else ""}: {", ".join(sections)}'
```
Comment (BoPeng) [inline src/sos/workflow_executor.py:1942]: THis one does not belong here.

File: src/sos/hosts.py [hunk 3]
```diff
@@ -536,17 +534,15 @@ def _map_var(self, source):
                 # pick the longest key that matches
                 k = max(matched, key=len)
                 dest = self.path_map[k] + dest[len(k):]
-            else:
```
[the remark this answers is not in the document]
Comment (jenningsje) [reply src/sos/hosts.py:539]: I'll start fixing that right now.

**Review by BoPeng — APPROVED** [review]

**Pull request merged**